packages/flutter/test/material/menu_anchor_test.dart

```diff
@@ -4947,6 +4947,36 @@ void main() {
       expect(find.byIcon(disabledIcon), findsOneWidget);
     });
   });
+
+  testWidgets('MenuAnchor.of gives nearest MenuAnchor ancestor', (WidgetTester tester) async {
```

Comment:
This needs some more tests:
1) MenuAnchor.of() throws an assertion error if there is no menu controller for a given context
2) MenuAnchor.maybeOf() can return null if there is no MenuController
3) MenuAnchor.maybeOf() can return the MenuController